### PR TITLE
[stable/external-dns] Enable assume-role creds for aws

### PR DESCRIPTION
**Pull request by jvassev**
Aws credentials are now stored in ~/.aws/, mounted from a secret.
This layout enables specifying a role_arn in case assume roles
creds are used. Update README with the new value `aws.roleArn`.

Also add image.pullSecrets in case an image needs to be customized.

### CODE REVIEW

File: stable/external-dns/templates/_helpers.tpl [hunk 1]
```diff
@@ -28,3 +28,17 @@ release: {{.Release.Name }}
 {{ toYaml .Values.podLabels }}
 {{- end }}
 {{- end }}
+
+{{- define "aws-credentials" }}
```
Comment (unguiculus) [inline stable/external-dns/templates/_helpers.tpl]: `{{- define "external-dns.aws-credentials" }}`

File: stable/external-dns/templates/_helpers.tpl [hunk 2]
```diff
@@ -28,3 +28,17 @@ release: {{.Release.Name }}
...
+[default]
+aws_access_key_id = {{ .Values.aws.accessKey }}
+aws_secret_access_key = {{ .Values.aws.secretKey }}
+{{ end }}
+
+
+{{- define "aws-config" }}
```
Comment (unguiculus) [inline stable/external-dns/templates/_helpers.tpl]: `{{- define "external-dns.aws-config" }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please make named templates namespaced.

File: stable/external-dns/templates/secret.yaml [hunk 3]
```diff
@@ -7,10 +7,10 @@ metadata:
   name: {{ template "external-dns.fullname" . }}
 type: Opaque
 data:
-{{- if and .Values.aws.secretKey .Values.aws.accessKey }}
-  aws_access_key_id: {{ .Values.aws.accessKey | b64enc | quote }}
-  aws_secret_access_key: {{ .Values.aws.secretKey | b64enc | quote }}
-{{- end }}
+{{- if eq .Values.provider "aws" }}
+  credentials: {{include "external-dns.aws-credentials" . | b64enc | quote}}
```
Comment (unguiculus) [inline stable/external-dns/templates/secret.yaml]: Space after `{{` and before `}}`, please.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/external-dns/templates/secret.yaml (hunk 3 above)
Comment (jvassev) [reply stable/external-dns/templates/secret.yaml]: Done, I also added missing spaces in other templates.

**Comment by unguiculus** [comment]
cc @linki

**Comment by linki** [comment]
lgtm

**Review by unguiculus — APPROVED** [review]

**Comment by jvassev** [comment]
/retest

I had to rebase and bump Chart version again to 0.5.1

**Pull request merged**